static/js/util/util.js

```diff
@@ -280,3 +292,36 @@ export function makeStrippedHtml(textOrElement) {
     return striptags(textOrElement);
   }
 }
+
+/**
+ * Validate a month number
+ * @param {String} string The input string
+ * @returns {Number|undefined} The valid month or undefined if not valid
+ */
+export function validateMonth(string) {
+  let month = filterPositiveInt(string);
+  if (month === undefined) {
+    return undefined;
+  }
+  if (month < 1 || month > 12) {
+    return undefined;
+  }
+  return month;
+}
+
+/**
+ * Validate a year string is an integer and fits into YYYY
+ * @param {String} string The input string
+ * @returns {Number|undefined} The valid year or undefined if not valid
+ */
+export function validateYear(string) {
+  let year = filterPositiveInt(string);
```

Comment:
For the `"2e0"` case that would become `20` since it just strips out non-numeric characters